controllers/space/nstemplatetier_spaces_mapper.go

```diff
@@ -0,0 +1,42 @@
+package space
+
+import (
+	"context"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/host-operator/controllers/nstemplatetier"
+
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var mapperLog = ctrl.Log.WithName("MapNSTemplateTierToSpaces")
+
+func MapNSTemplateTierToSpaces(namespace string, cl client.Client) func(object client.Object) []reconcile.Request {
+	return func(obj client.Object) []reconcile.Request {
+		if tmplTier, ok := obj.(*toolchainv1alpha1.NSTemplateTier); ok {
+			matchOutdated, err := nstemplatetier.OutdatedTierSelector(tmplTier)
+			if err != nil {
+				mapperLog.Error(err, "cannot map NSTemplateTier to Spaces")
```

Comment:
I'm fine with making the changes you suggested, but honestly, I don't see how they are better than what's in the PR 🤷‍♂️

Reply:
there were two changes in my suggestion:
1. I modified the text of the message - the original code used the same error message for all error cases, so I thought that it would be better to add more context in which cases it failed. 
It was only a suggestion - you didn't have to apply it if you didn't want to :wink: This explanation/answer would be enough :+1: 
2. I added the `return nil` line which is important from my point of view - to end the logic when there is an error